### ISSUE TITLE
Calling Hash#merge with no arguments or multiple arguments fails

### ISSUE THREAD

**Issue by mojavelinux** [opal/opal#1960]
Opal fails to merge a Hash when no arguments are given:

```rb
h = { a: 'a' }
h.merge
```

It also fails when multiple arguments are given:

```rb
h = { a: 'a' }
h.merge({ b: 'b' }, { c: 'c' })
```

It also fails if a single empty Hash is given and the argument list is not surrounded in brackets:

```rb
h = { a: 'a' }
h.merge {}
```

It works if a single empty Hash is given and the argument is surrounded in brackets (which indicates the previous example is a parser problem):

```rb
h = { a: 'a' }
h.merge({})
```

All of these succeed in CRuby 2.6 (as this is a 2.6 feature). The same examples fail in JRuby and CRuby < 2.6.

See http://ruby-doc.org/core-2.6/Hash.html#method-i-merge

**Comment by mojavelinux** [comment]
Here's an implementation:

```rb
Hash.prepend(Module.new do
  def merge *args
    args.length > 1 ? args.inject(self) {|acc, arg| acc.merge arg } : (super args.fetch 0, {})
  end
end) if (Hash.instance_method :merge).arity == 1
```

**Comment by elia** [comment]
Fixed by 4d93700d5 / #2187
